src/hotspot/os/linux/os_linux.cpp

```diff
@@ -79,37 +79,38 @@
 #include "utilities/vmError.hpp"
 
 // put OS-includes here
-# include <sys/types.h>
-# include <sys/mman.h>
-# include <sys/stat.h>
-# include <sys/select.h>
-# include <pthread.h>
-# include <signal.h>
+# include <dlfcn.h>
 # include <endian.h>
 # include <errno.h>
-# include <dlfcn.h>
+# include <fcntl.h>
+# include <fenv.h>
+# include <inttypes.h>
+# include <link.h>
+# include <linux/elf-em.h>
+# include <poll.h>
+# include <pthread.h>
+# include <pthread.h>
+# include <pwd.h>
+# include <signal.h>
+# include <stdint.h>
 # include <stdio.h>
+# include <string.h>
+# include <syscall.h>
 # include <unistd.h>
+# include <sys/ioctl.h>
+# include <sys/ipc.h>
+# include <sys/mman.h>
 # include <sys/resource.h>
-# include <pthread.h>
+# include <sys/select.h>
+# include <sys/shm.h>
+# include <sys/socket.h>
+# include <sys/stat.h>
 # include <sys/stat.h>
+# include <sys/sysinfo.h>
 # include <sys/time.h>
 # include <sys/times.h>
+# include <sys/types.h>
 # include <sys/utsname.h>
```

Comment:
> Do we need to shuffle^W sort includes in this patch? I presume you'd want this patch to be cleanly backportable, which means it should probably be as point-y as it can get.

OK, I can back that part out.